### ISSUE TITLE
No deserializer?

### ISSUE THREAD

**Issue by BHare1985** [adamfisher/RestSharp.Serializers.Newtonsoft.Json#8]
Is there a specific reason why you don't also include a deserializer as well? 
Would be nice to get an all in one solution.

Specifically, I don't like how the default RestSharp deserializer doesn't support enums the same way as json.net

**Comment by Rolorob** [comment]
I've got the same problem here, it would be helpful if deserialization is supported as well!

**Comment by quetzalcoatl** [comment]
The `public class RestRequest : IRestRequest` from RestSharp does not provide any way for that.
However, it seems that RestClient may be reconfigured to use a different serializers for each response mime type - see i.e. https://bytefish.de/blog/restsharp_custom_json_serializer/#using-the-custom-deserializer-for-incoming-responses or on github https://github.com/bytefish/bytefish.de/blob/master/blog/restsharp_custom_json_serializer.md

**Comment by quetzalcoatl** [comment]
In case the link rots, here's the code snippet from that article, with a tiny change in the name of implemented interfaces (since RestSharp changed a bit since then)

```
// Copyright (c) Philipp Wagner. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using Newtonsoft.Json;
using RestSharp.Serializers;
using System.IO;
using RestSharp.Deserializers;

namespace GcmSharp.Serialization
{
    public class NewtonsoftJsonSerializer : ISerializer, IDeserializer
    {
        private Newtonsoft.Json.JsonSerializer serializer;

        public NewtonsoftJsonSerializer(Newtonsoft.Json.JsonSerializer serializer)
        {
            this.serializer = serializer;
        }

        public string ContentType
        {
            get { return "application/json"; } // Probably used for Serialization?
            set { }
        }

        public string DateFormat { get; set; }

        public string Namespace { get; set; }

        public string RootElement { get; set; }

        public string Serialize(object obj)
        {
            using (var stringWriter = new StringWriter())
            {
                using (var jsonTextWriter = new JsonTextWriter(stringWriter))
                {
                    serializer.Serialize(jsonTextWriter, obj);

                    return stringWriter.ToString();
                }
            }
        }

        public T Deserialize<T>(RestSharp.IRestResponse response)
        {
            var content = response.Content;

            using (var stringReader = new StringReader(content))
            {
                using (var jsonTextReader = new JsonTextReader(stringReader))
                {
                    return serializer.Deserialize<T>(jsonTextReader);
                }
            }
        }

        public static NewtonsoftJsonSerializer Default
        {
            get
            {
                return new NewtonsoftJsonSerializer(new Newtonsoft.Json.JsonSerializer()
                {
                    NullValueHandling = NullValueHandling.Ignore,
                });
            }
        }
    }
}
```

and how to use it:

```
    var jser = Newtonsoft.Json.JsonSerializer.CreateDefault();
    // above: use any way to get the serializer, set its options, etc

    var wrapper = new GcmSharp.Serialization.NewtonsoftJsonSerializer(jser);

    request.JsonSerializer = wrapper;

    client.AddHandler("application/json", wrapper);
    client.AddHandler("text/json", wrapper);
    client.AddHandler("text/x-json", wrapper);
    client.AddHandler("text/javascript", wrapper);
    client.AddHandler("*+json", wrapper);
    // and so on for other MIME types you want to process
```

**Comment by Rolorob** [comment]
Thanks, I handled this the same way as you described but forgot to feed this back, sorry :) Thanks for your response, hope it helps others. Maybe I'll submit a PR later.

**Comment by adamfisher** [comment]
This has been added in version 1.5.0 thanks to the code @quetzalcoatl provided 👍. 

https://github.com/adamfisher/RestSharp.Serializers.Newtonsoft.Json/releases/tag/1.5.0

**Comment by adamfisher** [comment]
It seems this was also addressed with PR #4 but I forgot about it after approving it 😢
